MM-30515. Updated BIND parser.

rachelbritton opened the pull request:
Updated BIND parser reference article. Please review to make sure the content is correct and restructuring makes sense.

rachelbritton committed:
MM-30515. Updated BIND parser.

Updated BIND parser reference article.

rachelbritton committed:
MM-30515. Grammar/formatting edits.

Fixed some grammar and formatting inconsistencies.

Checked content against https://bluecatnetworks.atlassian.net/browse/MM-30144 to make sure all the customer's questions are addressed.

Martin-Metz commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,205 @@
-When ``named.conf`` is split up, a new file is created for each zone statement (see :ref:`bind-zone-options-files` files, below).
+1. ``named.conf``
+
+2. ``<BWF>/conf/``
```
Should be <BWD> for the Bind working directory, not BWF I guess.

Martin-Metz commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,205 @@
+.. note::
+   BWD is the BIND Working Directory, the location specified in the options directory statement. If no directory statement is found, the current directory is assumed unless you specify the root directory (-r argument). If the root directory is specified, Micetro will assume relative to the root.
+
+When started, the parser detects whether the directory ``<BWF>/conf/zones`` exists and whether the initial configuration file (typically ``named.conf``) contains one of the following phrases:
```
Same here BWD instead of BWF

Martin-Metz commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,205 @@
+* "generated by Micetro"
 
-The current version of the Men&Mice Management Console has no interface to change the values of the following statements: ``key``, ``acl``, ``controls``, ``server``, ``trusted-keys``. Therefore, they are kept in separate files, ``user_before`` and ``user_after``. Micetro does *not* modify these files.
+If both of these criteria are met, the parser assumes that configuration has been arranged and behaves accordingly.
```
The "arrange" command of the installer behaves accordingly not the parser, right? @indridib

indridib replied:
No it's the parser that checks for those criteria's and behaves according to that.

Martin-Metz commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,205 @@
+5. /etc/local/sbin
+6. /etc/opt/isc/isc-bind
+7. /etc/opt/isc/scls/isc-bind
+
```
Should we add a note on how to overwrite/specify a different path? I.e. by adding an XML-tag for each statement
`<named-checkconf value="/path to the /named-checkconf"/>`
`<named-checkzone value="/path to the /named-checkzone"/>`

Should we mention also that it is the preferences.cfg file of the DNS Server Controller agent?

rachelbritton replied:
@indridib

indridib replied:
Yes we could mention that. We are already mentioning named-checkconf elsewhere in the documentation https://docs.menandmice.com/en/latest/guides/reference/named-conf_location/#named-conf-validation 
We might want to update that and add explanation the named-checkzone preference value there and simply say here it can be overridden or set if needed, see: ....

indridib commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,204 @@
 .. meta::
    :description: When Micetro is installed on a server running BIND, it needs to perform several changes to the server configuration file structure. This section describes these changes and includes an overview diagram that shows how the server configuration files are processed
```
```suggestion
   :description: When Micetro is installed on a server running BIND, it will perform several changes to the server configuration file structure. This section describes these changes and includes an overview diagram that shows how the server configuration files are processed
```

indridib replied:
It doesn't need to any more, but it will by default arrange the files.

indridib commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,204 @@
-``named.conf`` is split into several files. A new ``named.conf`` file that includes statements for all the new files is created.
+Arranging the Configuration and Files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The parser's arranger splits the configuration into two files:
```
It's not two files. It's a single file named.conf and then a directory containing the configuration

indridib commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,204 @@
-When ``named.conf`` is split up, a new file is created for each zone statement (see :ref:`bind-zone-options-files` files, below).
+1. ``named.conf``
+
+2. ``<BWF>/conf/``
```
```suggestion
2. ``<BWD>/conf/``
```

indridib commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,204 @@
+.. note::
+   BWD is the BIND Working Directory, the location specified in the options directory statement. If no directory statement is found, the current directory is assumed unless you specify the root directory (-r argument). If the root directory is specified, Micetro will assume relative to the root.
+
+When started, the parser detects whether the directory ``<BWF>/conf/zones`` exists and whether the initial configuration file (typically ``named.conf``) contains one of the following phrases:
```
```suggestion
When started, the parser detects whether the directory ``<BWD>/conf/zones`` exists and whether the initial configuration file (typically ``named.conf``) contains one of the following phrases:
```

Martin-Metz commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,205 @@
+* "// This file was generated by"
 
-The options statement goes into a separate file. That file is modified when the user makes changes to the server options in the Men&Mice Management Console.
+* "generated by Micetro"
```
* `// This file was generated by the Men & Mice Suite`
or
* `// This file was generated by Micetro`

Martin-Metz commented on guides/reference/bind_file_structure.rst:
```diff
@@ -1,53 +1,205 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To determine whether a zone is a dynamic zone, the parser searches for two options:
+
+1. **allow-update**: Can be defined in option, view, and zone scope level. However, according to the `BINDv9Arm <https://bind9.readthedocs.io/en/v9.16.27/reference.html>`_, this option should only be set at the zone level. Setting a default value at the options or view level may result in some zones allowing unintentional updates.
```
We should reference the Bind 9.18 docs not 9.16 anymore:
https://bind9.readthedocs.io/en/bind-9.18/reference.html

Martin-Metz reviewed (approved):
Should we not replace all `/var/named` with `<BWD>`
as the `/var/named` is usually just a widely used example for the Bind working directory.
Please see also my inline comments, where suggested some changes/additions.

rachelbritton committed:
MM-30515. Revised reference articles.

Revised Bind File Structure based on feedback and added section to named.conf validation about overriding paths.

indridib commented:
Nice work, Can't think of anything else :-)

indridib reviewed (approved):
Nice work :-)
